test/boost/sstable_compaction_test.cc

```diff
@@ -2176,72 +2177,17 @@ class scrub_test_framework {
     }
 };
 
-SEASTAR_TEST_CASE(sstable_scrub_validate_mode_test) {
-    cql_test_config test_cfg;
+SEASTAR_THREAD_TEST_CASE(sstable_scrub_validate_mode_test) {
+    scrub_test_framework test;
 
-    auto& db_cfg = *test_cfg.db_config;
+    auto schema = test.schema();
 
-    // Disable cache to filter out its possible "corrections" to the corrupt sstable.
-    db_cfg.enable_cache(false);
-    db_cfg.enable_commitlog(false);
-
-    return do_with_cql_env([] (cql_test_env& cql_env) -> future<> {
-        return test_env::do_with_async([] (test_env& env) {
-            auto schema = schema_builder("ks", get_name())
-                    .with_column("pk", utf8_type, column_kind::partition_key)
-                    .with_column("ck", int32_type, column_kind::clustering_key)
-                    .with_column("s", int32_type, column_kind::static_column)
-                    .with_column("v", int32_type).build();
-            auto permit = env.make_reader_permit();
-
-            auto scrubbed_mt = make_lw_shared<replica::memtable>(schema);
-            auto sst = env.make_sstable(schema);
+    auto muts = tests::generate_random_mutations(test.random_schema(), 3).get();
+    std::swap(*muts.begin(), *(muts.begin() + 1));
 
-            testlog.info("Writing sstable {}", sst->get_filename());
-
-            const auto corrupt_fragments = write_corrupt_sstable(env, *sst, permit, [&, mut_builder = mutation_rebuilder_v2(schema)] (mutation_fragment_v2&& mf, bool) mutable {
-                if (mf.is_end_of_partition()) {
-                    scrubbed_mt->apply(*std::move(mut_builder).consume_end_of_stream());
-                } else {
-                    std::move(mf).consume(mut_builder);
-                }
-            });
-
-            sst->load(sst->get_schema()->get_sharder()).get();
-
-            testlog.info("Loaded sstable {}", sst->get_filename());
-
-            auto table = env.make_table_for_tests(schema);
-            auto close_cf = deferred_stop(table);
-            table->start();
-
-            table->add_sstable_and_update_cache(sst).get();
-
-            bool found_sstable = false;
-            foreach_table_state_with_thread(table, [&] (compaction::table_state& ts) {
-                auto sstables = in_strategy_sstables(ts);
-                if (sstables.empty()) {
-                    return;
-                }
+    test.run(schema, muts, [] (table_for_tests& table, compaction::table_state& ts, std::vector<sstables::shared_sstable> sstables) {
                 BOOST_REQUIRE(sstables.size() == 1);
```

Comment:
Is this BOOST_REQUIRE part of the test, or just an assumption you make on the test framework? I'm guessing it's the latter, but am I right?

Reply:
Yes, the latter.